x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/logical/Or.java

```diff
@@ -57,4 +59,9 @@ protected Expression canonicalize() {
         // NB: this add a circular dependency between Predicates / Logical package
         return Predicates.combineOr(Predicates.splitOr(super.canonicalize()));
     }
+
+    @Override
+    public boolean translatable(LucenePushdownPredicates pushdownPredicates) {
+        return super.translatable(pushdownPredicates) && FullTextFunction.checkDisjunctionPushable(this);
```

Comment:
Does this introduce a side effect where we might not pushdown a disjunction that could be pushed because it uses a FTF? I haven't tested this, but assuming we have:

```
FROM my-index
| WHERE match(foo, "bar") or x == "foobar"
```

Normally assuming that `x == "foobar"` can be pushed, we should be able to push `match(foo, "bar") or x == "foobar"` but I wonder if this still happens.
We added this method because we need it for the check where disjunctions are used with scoring.
I'd actually be in favour of not overriding this method here and just have the check done only in `FullTextFunction` if possible. It's also awkward how this method is actually called in `FullTextFunction` and then here we end up calling `FullTextFunction.checkDisjunctionPushable(this)`.

Reply:
> Does this introduce a side effect where we might not pushdown a disjunction that could be pushed because it uses a FTF?

You are correct! There is no longer a need to check disjunctions when scoring is not there. That's much better done in `FullTextFunction` class.

I corrected this in d23998d2eb4b761ea847aaaf5e6a0e4f97216786 and added a plan optimizer test to check that a disjunction with pushable functions is effectively pushed down.

Thank you!